test/e2e/flask/multichain-api/wallet_sessionChanged.spec.ts

```diff
@@ -0,0 +1,120 @@
+  largeDelayMs,
+  WINDOW_TITLES,
+  withFixtures,
+} from '../../helpers';
+import { Driver } from '../../webdriver/driver';
+import FixtureBuilder from '../../fixture-builder';
+import {
+  addAccountInWalletAndAuthorize,
+  DEFAULT_MULTICHAIN_TEST_DAPP_FIXTURE_OPTIONS,
+  getExpectedSessionScope,
+  initCreateSessionScopes,
+  openMultichainDappAndConnectWalletWithExternallyConnectable,
+  updateNetworkCheckboxes,
+} from './testHelpers';
+
+describe('Call `wallet_createSession`, then update the accounts and/or scopes in the permissions page of the wallet for that dapp', function () {
+  const INITIAL_SCOPES = ['eip155:1337', 'eip155:1338'];
+  const REMOVED_SCOPE = INITIAL_SCOPES[0];
+  const UPDATED_SCOPE = INITIAL_SCOPES[1];
+
+  const ACCOUNTS = [ACCOUNT_1, ACCOUNT_2];
+  const UPDATED_ACCOUNT = ACCOUNTS[1];
+  it('should receive a `wallet_sessionChanged` event with the full new session scopes', async function () {
+    await withFixtures(
+      {
+        title: this.test?.fullTitle(),
+        fixtures: new FixtureBuilder()
+          .withNetworkControllerTripleGanache()
+          .build(),
+        ...DEFAULT_MULTICHAIN_TEST_DAPP_FIXTURE_OPTIONS,
+      },
+      async ({
+        driver,
+        extensionId,
+      }: {
+        driver: Driver;
+        extensionId: string;
+      }) => {
+        await openMultichainDappAndConnectWalletWithExternallyConnectable(
+          driver,
+          extensionId,
+        );
+        await initCreateSessionScopes(driver, INITIAL_SCOPES, ACCOUNTS);
+        await addAccountInWalletAndAuthorize(driver);
+        await driver.clickElement({ text: 'Connect', tag: 'button' });
+        await driver.delay(largeDelayMs);
+        await driver.switchToWindowWithTitle(
+          WINDOW_TITLES.ExtensionInFullScreenView,
+        );
+
+        /**
+         * We make sure to update selected accounts via wallet extension UI
+         */
+        await driver.clickElementSafe(
+          '[data-testid="account-options-menu-button"]',
+        );
+        await driver.clickElementSafe(
+          '[data-testid="global-menu-connected-sites"]',
+        );
+        await driver.clickElementSafe('[data-testid="connection-list-item"]');
+
+        const editButtons = await driver.findElements('[data-testid="edit"]');
+        await editButtons[0].click();
+        const checkboxes = await driver.findElements(
+          'input[type="checkbox" i]',
+        );
+        const firstAccountCheckbox = checkboxes[1];
+        await firstAccountCheckbox.click();
+        await driver.clickElementSafe({ text: 'Update', tag: 'button' });
+
+        /**
+         * And also update selected scope to {@link UPDATED_SCOPE}
+         */
+        await updateNetworkCheckboxes(driver, ['Localhost 8546']);
+        await driver.switchToWindowWithTitle(WINDOW_TITLES.MultichainTestDApp);
+
+        const walletSessionChangedNotificationWebElement =
+          await driver.findElement('#wallet-session-changed-result-0');
```

Comment:
just want to make sure that initial page connection via `initCreateSessionScopes` doesn't cause a wallet_sessionChanged to be populated here. I expect it does, and if it does, then should we be checking `result-1` here instead?